src/cloudant/database.py

```diff
@@ -152,15 +152,12 @@ def create_document(self, data, throw_on_exists=False):
             corresponding to the new document in the database.
         """
         docid = data.get('_id', None)
-        doc = None
         if docid and docid.startswith('_design/'):
             doc = DesignDocument(self, docid)
         else:
             doc = Document(self, docid)
-        if throw_on_exists and doc.exists():
-            raise CloudantDatabaseException(409, docid)
         doc.update(data)
-        doc.create()
+        doc.create(throw_on_exists=throw_on_exists)
         super(CouchDatabase, self).__setitem__(doc['_id'], doc)
```

Comment:
Good point about the `CloudantDatabaseException`, I guess in my suggestion above rather than re-raising if throw_on_exist is True instead raise the `CloudantDatabaseException` 409 while everything else remains the same.  I'm not sure that I follow your poiont on the HEAD check though.